includes/core/assets/js/event-deny-list.js

```diff
@@ -0,0 +1,7 @@
+/**
+ * Remove unwanted blocks from given post type.
+ */
+
+wp.domReady(function () {
+	wp.blocks.unregisterBlockType('gatherpress/events-list');
```

Comment:
I would add `venue-information` to this one and leave `events-list` to be allowed on an event page.